retsinformation.js

```diff
@@ -0,0 +1,182 @@
+		newItem.shortTitle = getRessort(doc, url)[0];
+	} else if (type=="report") {
+		newItem = addAuthor (newItem, getRessort(doc, url)[1]);
+		newItem.date = getRessort(doc, url)[2];
+		newItem.institution = "Folketinget";
+		threeLetters = getKortNavn(doc, url)[0].substr(0,3);
+		if (threeLetters=="EDP") {
+			newItem.seriesTitle = "Folketingets Ombudsmands driftundersøgelser";
+		} else if (threeLetters=="ISP") {
+			newItem.seriesTitle = "Folketingets Ombudsmands Inspektioner";
+		} else if (threeLetters=="FOU") {
+			newItem.seriesTitle = "Folketingets Ombudsmands udtalelser";
+		};
+		newItem.reportNumber = getKortNavn(doc, url)[0];
+	} else if (type=="bill") {
+		newItem = addAuthor (newItem, getRessort(doc, url)[1]);
+		newItem.billNumber = getKortNavn(doc, url)[0];
+		newItem.date = getRessort(doc, url)[2];
+	} else {
+		return;
+	};
+	newItem.complete();
+	return;
+}
+
+function getTitle (doc, url) {
+	var myXPath = '//div[@class="wrapper2"]/div/p[@class="Titel2"]';
+	title = ZU.xpathText(doc, myXPath);
+	if (!title) {
+		myXPath = '//div[@class="wrapper2"]/div/div[@id="INDHOLD"]/p[@class="Titel"]';
+		title = ZU.xpathText(doc, myXPath);
+	};
+	if (!title) {
+		myXPath = '//div[@class="wrapper2"]/div/div[@id="INDHOLD"]/font/p[@align="CENTER"]';
+		title = ZU.xpathText(doc, myXPath);
+	};
+	if (!title) {
+		myXPath = '//div[@class="wrapper2"]/div/div[@id="INDHOLD"]/h1[@class="TITLE"]';
+		title = ZU.xpathText(doc, myXPath);
+	};
+	if (!title) {
+		myXPath = '//div[@class="wrapper2"]/div/div[@id="INDHOLD"]/center/h1[@class="TITLE"]';
+		title = ZU.xpathText(doc, myXPath);
+	};
+	if (!title) {
+		var myXPath = '/html/head/title';
+		title = ZU.xpathText(doc, myXPath);
+		title = title.substring(0,title.lastIndexOf("-"));
+		if (getRessort(doc, url)[0]) {
+			title = title.substring(getRessort(doc, url)[0].length+4,title.length);
+		};
+	};
+	title = clean(title);
+	Z.debug("L111"+title);
+	return title;
+}
+
+function getKortNavn (doc, url) {
+	var myXPath = '//span[@class="kortNavn"]';
+	if (!ZU.xpathText(doc, myXPath)) {
+		myXPath = '//span[@class="kortNavn historisk"]';
+	};
+	infoArray = ZU.xpathText(doc, myXPath);
+	infoArray = infoArray.replace(/ (Gældende|Historisk)/,"");
+	infoArray = infoArray.split(" af ");
+	return infoArray;
+}
+
+function getRessort (doc, url) {
+	var myXPath = '//div[@class="ressort"]';
+	fodder = ZU.xpathText(doc, myXPath);
+	fodder = fodder.trim();
+	if (fodder.slice(0,1) == "(") {
+		infoArray[0] = fodder.substr(1,fodder.indexOf(")")-1);
+	} else {
+		infoArray[0] = "";
+	};
+	pubDate = fodder.search(/\d{2}\-\d{2}\-\d{4}/);
+	infoArray[1] = fodder.substring(pubDate+10,fodder.length);
+	infoArray[2] = fodder.substr(pubDate,10);
+	Zotero.debug("L140"+infoArray);
+	return infoArray;
+}
+
+function addAuthor (item, author) {
+	item.creators[0] = {};
+	item.creators[0].creatorType = "author";
+	item.creators[0].lastName = author;
+	item.creators[0].fieldMode = 1;
+	return item;
+}
```

Comment:
I tried to not return item - it didn't work.

I made this function to avoid writing the code multiple times. I can't see how to accomplish the same inside the scrape function.